docs/rules/no-setup-in-describe.md

```diff
@@ -0,0 +1,71 @@
+
+# Disallow setup in describe blocks (no-setup-in-describe)
+
+Setup for test cases in mocha should be done in `before`, `beforeEach`, or `it` blocks. Unfortunately there is nothing stopping you from doing setup directly inside a `describe` block.
+
+```js
+describe('something', function () {
+    const a = setup();
+    const b = a.getter;
+    it('should work', function () {
+        assert(a === b);
+    });
+});
+```
+
+Any setup directly in a `describe` is run before all tests execute. This is undesirable primarily for two reasons:
```

Comment:
What about top-level code outside of a `describe`?

Reply:
I'd love to handle this, but it could result in worse performance on non-test code (currently a `.length check` is all the overhead that is incurred outside a `describe`). Thoughts on that tradeoff?

Reply:
Hmm, actually I just don't know how to flag a file as having a `describe` with eslint. What's the perf on a `hasDescribe()` and can this even be done?

Reply:
I’m not sure why it should be necessary to check if there is a `describe` in a specific file, you could just look for setup code directly in the `Program` node.
Anyway I’m not sure if this is wanted, because most people probably prefer to have their `require` calls at the top of the file. So maybe we leave it for now as it is and if some one requests that feature we can implement this behind an option.